# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLExtensions.cpp
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 dom/canvas/WebGLExtensions.cpp
--- a/dom/canvas/WebGLExtensions.cpp	Wed Jan 08 21:52:03 2020 +0000
+++ b/dom/canvas/WebGLExtensions.cpp	Wed Jan 08 22:19:14 2020 +0000
@@ -13,7 +13,7 @@
 namespace mozilla {
 
 WebGLExtensionBase::WebGLExtensionBase(WebGLContext* webgl)
-    : WebGLContextBoundObject(webgl), mIsLost(false), mIsExplicit(false) {}
+    : WebGLContextBoundObject(webgl) {}
 
 WebGLExtensionBase::~WebGLExtensionBase() {}
 
@@ -23,11 +23,6 @@
   OnMarkLost();
 }
 
-NS_IMPL_CYCLE_COLLECTION_WRAPPERCACHE_0(WebGLExtensionBase)
-
-NS_IMPL_CYCLE_COLLECTION_ROOT_NATIVE(WebGLExtensionBase, AddRef)
-NS_IMPL_CYCLE_COLLECTION_UNROOT_NATIVE(WebGLExtensionBase, Release)
-
 // -
 
 WebGLExtensionExplicitPresent::WebGLExtensionExplicitPresent(
@@ -46,8 +41,6 @@
   mContext->PresentScreenBuffer();
 }
 
-IMPL_WEBGL_EXTENSION_GOOP(WebGLExtensionExplicitPresent, WEBGL_explicit_present)
-
 // -
 
 WebGLExtensionFloatBlend::WebGLExtensionFloatBlend(WebGLContext* const webgl)
@@ -68,8 +61,6 @@
   return gl->IsExtensionSupported(gl::GLContext::EXT_float_blend);
 }
 
-IMPL_WEBGL_EXTENSION_GOOP(WebGLExtensionFloatBlend, EXT_float_blend)
-
 // -
 
 WebGLExtensionFBORenderMipmap::WebGLExtensionFBORenderMipmap(
@@ -90,8 +81,6 @@
   return gl->IsExtensionSupported(gl::GLContext::OES_fbo_render_mipmap);
 }
 
-IMPL_WEBGL_EXTENSION_GOOP(WebGLExtensionFBORenderMipmap, OES_fbo_render_mipmap)
-
 // -
 
 WebGLExtensionMultiview::WebGLExtensionMultiview(WebGLContext* const webgl)
@@ -123,6 +112,4 @@
                                         baseViewIndex, numViews);
 }
 
-IMPL_WEBGL_EXTENSION_GOOP(WebGLExtensionMultiview, OVR_multiview2)
-
 }  // namespace mozilla